iroha/src/torii.rs

```diff
@@ -393,6 +393,13 @@ async fn handle_request(
                 let version = message.version();
                 if let Some(message) = message.into_v1() {
                     let message: SumeragiMessage = message.into();
+                    if let Err(e) =
+                        message.check_instruction_len(state.read().await.max_instruction_number)
```

Comment:
For example, can we create `ValidSumeragiMessage` which will be created via `SumeragiMessage::validate` and inside `validate` message we will call `Block::validate`? In this case we only need to know that message is valid. What actions should be performed to validate message will be encapsulated inside message, which also will only know that it should validate blocks, and `Block::validate` will know about `Transaction::validate` and only on transaction layer I see a need to check instructions length (it can be private method btw for transactions module only that way).